### DISCUSSION TITLE
Promice gets called twice (first with empty result, then with result)

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#4346] [1. Help] [answered]
I have the following code:

Trigger

>           this.$watch('search.inspectFlick', flick => {
>             console.warn({flick})
>             if(flick)
>               this.loadCredits(flick).then(c => this.setPersons(c));
>           });

Function

>     setPersons(credits){
>        console.warn({ credits })    
>     }

Which leads to this console output:
![image](https://github.com/user-attachments/assets/d0a045ec-2437-4720-95e0-953271be4706)

As you can see, there is only one change in `search.inspectFlick`, but `setPersons()` gets called twice, once with empty credits and once with full credits. I get the `loadCredits()` returns a Promise, but shouldn't `then(...)` only be called when that promise gets fullfilled?

This way I always have to check if credits are not empty, isn't there a better way to only have 1 call of `setPersons` with the fullfilled promise?

**Comment by ekwoka** [comment, marked as answer]
This only makes sense if you have something else also calling setPersons.

What does load credits or whatever do?

It's showing two totally different value types.

One being an Object and the other being an array.

So something is going on elsewhere on your code.

**Comment by simonferndriger** [comment]
Ah sorry, you are totally right - it was called elsewhere with a different parameter (did not recognize that it was an array instead of object, I just thought it was proxied)
